src/java/boa/compiler/visitors/TypeCheckingVisitor.java

```diff
@@ -955,6 +955,9 @@ public void visit(final VarDeclStatement n, final SymbolTable env) {
 
 			if (lhs instanceof BoaArray && rhs instanceof BoaTuple)
 				rhs = new BoaArray(((BoaTuple)rhs).getMember(0));
+			if (lhs instanceof BoaArray && rhs instanceof BoaProtoList){
+				((BoaProtoList)rhs).setType(lhs);
```

Comment:
This is just wrong.  The proto list already has a type set (when it was created).  Why is this here?